crates/sui-cost-tables/src/tables.rs

```diff
@@ -0,0 +1,1120 @@
+            Opcodes::CALL_GENERIC,
+            ((ty_args.len() + args.len() + 1) as u64).into(),
+        )
+    }
+
+    fn charge_ld_const(&mut self, size: NumBytes) -> PartialVMResult<()> {
+        self.charge_instr_with_size(Opcodes::LD_CONST, u64::from(size).into())
+    }
+
+    fn charge_copy_loc(&mut self, val: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(Opcodes::COPY_LOC, val.legacy_abstract_memory_size())
+    }
+
+    fn charge_move_loc(&mut self, val: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(Opcodes::MOVE_LOC, val.legacy_abstract_memory_size())
+    }
+
+    fn charge_store_loc(&mut self, val: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(Opcodes::ST_LOC, val.legacy_abstract_memory_size())
+    }
+
+    fn charge_pack(
+        &mut self,
+        is_generic: bool,
+        args: impl ExactSizeIterator<Item = impl ValueView>,
+    ) -> PartialVMResult<()> {
+        let field_count = AbstractMemorySize::new(args.len() as u64);
+        self.charge_instr_with_size(
+            if is_generic {
+                Opcodes::PACK_GENERIC
+            } else {
+                Opcodes::PACK
+            },
+            args.fold(field_count, |acc, val| {
+                acc + val.legacy_abstract_memory_size()
+            }),
+        )
+    }
+
+    fn charge_unpack(
+        &mut self,
+        is_generic: bool,
+        args: impl ExactSizeIterator<Item = impl ValueView>,
+    ) -> PartialVMResult<()> {
+        let field_count = AbstractMemorySize::new(args.len() as u64);
+        self.charge_instr_with_size(
+            if is_generic {
+                Opcodes::UNPACK_GENERIC
+            } else {
+                Opcodes::UNPACK
+            },
+            args.fold(field_count, |acc, val| {
+                acc + val.legacy_abstract_memory_size()
+            }),
+        )
+    }
+
+    fn charge_read_ref(&mut self, ref_val: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(Opcodes::READ_REF, ref_val.legacy_abstract_memory_size())
+    }
+
+    fn charge_write_ref(&mut self, val: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(Opcodes::WRITE_REF, val.legacy_abstract_memory_size())
+    }
+
+    fn charge_eq(&mut self, lhs: impl ValueView, rhs: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(
+            Opcodes::EQ,
+            lhs.legacy_abstract_memory_size() + rhs.legacy_abstract_memory_size(),
+        )
+    }
+
+    fn charge_neq(&mut self, lhs: impl ValueView, rhs: impl ValueView) -> PartialVMResult<()> {
+        self.charge_instr_with_size(
+            Opcodes::NEQ,
+            lhs.legacy_abstract_memory_size() + rhs.legacy_abstract_memory_size(),
+        )
+    }
+
+    fn charge_load_resource(&mut self, _loaded: Option<NumBytes>) -> PartialVMResult<()> {
+        Ok(())
+    }
+
+    fn charge_borrow_global(
```

Comment:
Should we just invariant violation for all of these global storage operations?

Reply:
Will do. This includes `Exists`, `MoveFrom`, `MoveTo`, right?

Reply:
Correct